src/index.js

```diff
@@ -164,6 +164,10 @@ const setUpSocket = () => {
     store.connected = true;
     store.onConnectionChangeCallback(true);
 
+    if (store.socket.readyState !== WebSocket.OPEN) {
```

Comment:
This will fix the symptom but not the cause. This is inside socket open callback in which case the socket should already be in open state. 

I think that the scenario is more complicated involving two simultaneous socket sessions.
Say you visit a bot, the chat opens and the socket starts connecting. Then you switch the bot, I think it's somehow calling send on the previous socket reference most likely because send is called before the previous socket closed. 

You can try to replicate this in training page, use the dropdown to rapidly switch between bots.

Alternatively you can manually add a timeout for close.

Reply:
The hypothesis doesn't seem to be true, I tried different sort of delays, it seems that socket references are handled correctly (apart from maybe some rare inconsistent cases).

<img width="1232" alt="Screen Shot 2019-08-20 at 1 47 28 PM" src="https://user-images.githubusercontent.com/14262975/63346385-85762c80-c35d-11e9-954d-ff0355b9dbd7.png">

I spotted the problem a few times while testing, but couldn't find a way to reproduce it consistently.